src/agentscope_runtime/engine/deployers/utils/service_utils/fastapi_factory.py

```diff
@@ -693,9 +693,32 @@ def _create_streaming_parameter_wrapper(
         parsing."""
         is_async_gen = inspect.isasyncgenfunction(handler)
 
+        # NOTE:
+        # -----
+        # FastAPI >= 0.123.5 uses Dependant.is_coroutine_callable, which in
```

Comment:
Since this behavior relies on FastAPI’s internal implementation details and the `inspect.unwrap` logic, should we explicitly constrain the FastAPI version in **pyproject.toml** (e.g., `<=0.123.4` or `>=0.123.5`) to ensure consistent behavior across environments?

Alternatively, is the current implementation fully compatible with both pre‑0.123.5 and post‑0.123.5 versions, making a strict version pin unnecessary? If compatibility is confirmed, we might instead document the supported FastAPI versions in the README and release notes to guide future upgrades.